### ISSUE TITLE
MOEAD gets stuck for popsizes = 2

### ISSUE THREAD

**Issue by Sceki** [esa/pagmo2#318]
Hello,

I seem to have encountered a bug in `moead`. The problem seems to appear when only 2 individuals are chosen as `popsize`. The following minimal example reproduces it:
```
BOOST_AUTO_TEST_CASE(minimal_example)
{
    problem prob{zdt{1u, 30u}};
    population pop1{prob, 2u, 23u};

    moead user_algo1{100u, "grid", "tchebycheff", 1u, 1., 0.5, 20., 0.9, 2u, true, 23u};
    user_algo1.set_verbosity(1u);
    pop1 = user_algo1.evolve(pop1);
}
```

**Comment by bluescarni** [comment]
@Sceki any idea on where it gets stuck? I took a quick glance but I didn't see anything obvious popping up related to the number of individuals ``NP``.

**Comment by Sceki** [comment]
Well, my guess is that is inside the `while` cycle in the `select_private` method.. since the algo seems to do an infinite loop.. but I am not sure.. I will do a check now

**Comment by Sceki** [comment]
Indeed I confirm.. that loop is executed infinitely, thus causing the algorithm to get stuck..

**Comment by bluescarni** [comment]
So is the issue the fact that it's trying to extract 2 parents from a population of 2? Would bumping up the minimum population size for MOEAD from 2 to 3 solve the issue then?

**Comment by Sceki** [comment]
I am not sure if that will work.. This also presents the same issue:

```
BOOST_AUTO_TEST_CASE(minimal_example)
{
    problem prob{zdt{1u, 30u}};
    population pop1{prob, 3u, 23u};

    moead user_algo1{100u, "grid", "tchebycheff", 1u, 1., 0.5, 20., 0.9, 2u, true, 23u};
    user_algo1.set_verbosity(1u);
    pop1 = user_algo1.evolve(pop1);
}
```

**Comment by Sceki** [comment]
perhaps by putting the min pop size to 3, and the min neigh size to 2 would fix it.. I will check

**Comment by bluescarni** [comment]
Ok, thanks for checking this out! Pinging @darioizzo .

**Comment by Sceki** [comment]
I indeed think that enforcing `m_neighbours>1` and `popsize>=3` would fix the issue.. not sure if this can also be solved internally in the algo.. but I do not personally think it is even worthy to check inside the algo since a population size of 3 seems already small enough

**Comment by Sceki** [comment]
Let me know in case you want me to do a PR !

**Comment by darioizzo** [comment]
@Sceki Yes well spotted. Obviously https://github.com/esa/pagmo2/blob/master/src/algorithms/moead.cpp#L385 will never be `false` if popsize is only 2 as each individual will have a neighbourhood of size 1, and parents are selected from the neighbourhood.

As I see it the  ```popsize>=3``` should fix the bug. Why would you also change ```m_neighbours>1```?

**Comment by Sceki** [comment]
I believe that the problem is when `m_neighbours=1`. For instance if I test:
```
BOOST_AUTO_TEST_CASE(minimal_example)
{
    problem prob{zdt{1u, 30u}};
    population pop1{prob, 30u, 23u};

    moead user_algo1{100u, "grid", "tchebycheff", 1u, 1., 0.5, 20., 0.9, 2u, true, 23u};
    user_algo1.set_verbosity(1u);
    pop1 = user_algo1.evolve(pop1);
}

```
The bug is again present.

That is the reason why I would enforce `m_neighbours>1`

**Comment by darioizzo** [comment]
@Sceki You are right .... Please make a PR since you are on it. Also maybe add an assert in https://github.com/esa/pagmo2/blob/master/src/algorithms/moead.cpp#L375 to check that ss >=2 and neigh_idx.size() >=2 ?

**Comment by Sceki** [comment]
Sure!

I will do it asap

**Comment by bluescarni** [comment]
Fixed by #320.
